dftarea.cpp

```diff
@@ -311,6 +311,17 @@ cv::Mat DFTArea::grayComplexMatfromImage(QImage &img){
         for (unsigned int i = 0; i < igramArea->m_polygons[n].size(); ++i){
             int x = round((igramArea->m_polygons[n][i].x - left) * scaleFactor);
             int y = round((igramArea->m_polygons[n][i].y - top) * scaleFactor);
+
+            // make sure x and y values of regions are inside our matrix
+            if (x < 0)
+                x=0;
```

Comment:
This is only a suggestion:
```suggestion
            if (x < 0)
            {
                x=0;
            }
```
Matter of taste but not only. I believe this is MISRA rule 15.6 meaning enough people made mistakes with such code that defensive style is recommended for everybody.